frontend/src/reducer/user.js

```diff
@@ -46,11 +62,7 @@ export const defaultUser = {
 export default function user (state = defaultUser, action) {
   switch (action.type) {
     case `${SET}/${USER_CONNECTED}`:
-      return {
-        ...state,
-        ...serializeUser(action.user),
-        lang: serializeUser(action.user).lang || state.lang
-      }
+      return serializeUser({ ...unserializeUser(state), ...action.user })
```

Comment:
Would something like this work?

```suggestion
      return {...state, ...action.user }
```

Reply:
No, because `action.user` does not have all the properties of `state` and we want overwrite only the properties that it has, but when it goes through the `serializeUser` these properties are set as `undefined` and overwrite all the properties of `state`.

Reply:
Ah right. This is a bit frustrating to unserialize and then serialize almost the same thing and this looks like a code that could be rewritten without thinking about what you said. Could you add a comment to warn about this?

Reply:
Done